tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionSetMethodContract.scala

```diff
@@ -1520,4 +1526,289 @@ trait FirebaseSubscriptionSetMethodContract {
          |    ]
          |}""".stripMargin)
   }
+
+  @Test
+  def updateExpiresShouldReturnUpdatedWhenValidRequest(server: GuiceJamesServer): Unit = {
+    val firebaseSubscription1 = server.getProbe(classOf[FirebaseSubscriptionProbe])
+      .createSubscription(BOB, FIREBASE_SUBSCRIPTION_CREATE_REQUEST)
+
+    val newExpires = UTCDate(ZonedDateTime.now().plusDays(3)).asUTC.format(TIME_FORMATTER)
+
+    val updateResponse = `given`
+      .body(
+        s"""{
+           |  "using": [
+           |    "urn:ietf:params:jmap:core",
+           |    "com:linagora:params:jmap:firebase:push"],
+           |  "methodCalls": [[
+           |        "FirebaseRegistration/set",
+           |        {
+           |            "update": {
+           |                "${firebaseSubscription1.id.serialize}": {
+           |                  "expires": "$newExpires"
+           |                }
+           |              }
+           |        },
+           |    "c1"]]
+           |}""".stripMargin)
+    .when
+      .post
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(updateResponse).isEqualTo(
+      s"""{
+         |    "sessionState": "${SESSION_STATE.value}",
+         |    "methodResponses": [
+         |        [
+         |            "FirebaseRegistration/set",
+         |            {
+         |                "updated": {
+         |                    "${firebaseSubscription1.id.serialize}": {
+         |                      "expires": "$newExpires"
```

Comment:
As I remember, if the updated value is not diff from the input one, we will not return the updated value.